kimchi_state/src/navigation_manager.cpp

```diff
@@ -103,6 +103,7 @@ void NavigationManager::localizeResultCallback(
   switch (result.code) {
     case rclcpp_action::ResultCode::SUCCEEDED:
       RCLCPP_INFO(node_->get_logger(), "[NavigationManager] Goal succeeded!");
+      mission_observer_->onMissionFinished();
```

Comment:
Lets the state server know that the robot was localized so it can set the state to IDLE

Reply:
This is good for now here. But in an upcoming PR I'm addressing this issue a bit different!